ISPN-6333 - Adding more tests for Remote Script Execution

andyuk1986 opened the pull request:
ISPN-6333 - https://issues.jboss.org/browse/ISPN-6333
- Added more tests verifying Script execution;
- Enabled a few disabled tests based on bugfixes; 

Fixed ISPN-6302 -  https://issues.jboss.org/browse/ISPN-6302

slaskawi commented:
Hey @andyuk1986! Could you please rebase it? I would like to make sure we won't introduce new failures on CI.

andyuk1986 commented:
Hi @slaskawi :) Done!

slaskawi commented on server/integration/testsuite/src/test/java/org/infinispan/server/test/task/DistributedAuthServerTaskIT.java:
```diff
@@ -0,0 +1,121 @@
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldRunDistAuthTest() throws Exception {
```
It seems there is a failure: http://ci.infinispan.org/viewLog.html?buildId=37568&buildTypeId=bt9&guest=1#testNameId-1517987404531832027

slaskawi commented:
Thanks! It seems there is 1 related failure (I've checked the rest of them and they seem to be unrelated). The code looks good to me.

andyuk1986 commented:
@slaskawi I have fixed the test as well as all *ServerTaskIT tests by properly undeploying the deployed jar file. I have also reopend the DistributedServerTaskDeploymentIT test to see whether it will work with this solution. Waiting for CI build.

andyuk1986 commented:
@slaskawi I have fixed the failing test, as well as seems that DistributedServerTaskDeploymentIT test is not failing anymore - fixed with a separate commit. Also I have rebased with the latest changes. After that the testsuite was run twice on CI and no pull request related failures appeared.

slaskawi commented:
Looks good to me.

slaskawi commented:
Sorry, it's taking a bit long... Could you please rebase it once more?

andyuk1986 commented:
@slaskawi no problem :) Rebase is done :)

danberindei commented on scripting/src/test/java/org/infinispan/scripting/ClusteredScriptingTest.java:
```diff
@@ -2,89 +2,233 @@
-      assertEquals(resultsFuture.get(0).get("macbeth"), Long.valueOf(287));
-      assertEquals(resultsFuture.get(1).get("macbeth"), Long.valueOf(287));
+   @DataProvider(name = "cacheModeProvider")
+   private static Object[][] providePrinciples() {
```
Method name seems out of place

andyuk1986 replied:
Fixed!

andyuk1986 commented:
Rebase is done!

slaskawi commented:
hmmm there seems to be a lot of failures: http://ci.infinispan.org/viewLog.html?buildId=37714&buildTypeId=bt9&guest=1

I retriggered CI...

andyuk1986 commented:
@slaskawi the only failure related to my pull request is: DistributedServerTaskIT.shouldThrowExceptionInRemoteTasks It is because of changes done by Will related to ISPN-6253. I will rebase once more my code and fix this test if it is not fixed already in the upstream. The CI build will be triggered again so we can watch if there is something which still fails.

andyuk1986 commented:
Rebased!

andyuk1986 commented:
@slaskawi the CI build is finished. There are 3 failures not related to the pull request.

tristantarrant commented:
Thanks @andyuk1986 , pushed to master
